pentasharp/Services/DriverDashboardService.cs

```diff
@@ -0,0 +1,287 @@
+﻿using System.Collections.Generic;
+using System.Threading.Tasks;
+using AutoMapper;
+using Microsoft.EntityFrameworkCore;
+using pentasharp.Data;
+using pentasharp.Interfaces;
+using pentasharp.Models.Enums;
+using pentasharp.Models.TaxiRequest;
+using Microsoft.Extensions.Logging;
+
+
+namespace pentasharp.Services
+{
+
+    public class DriverDashboardService : IDriverDashboardService
+    {
+        private readonly IAuthenticateService _authService;
+        private readonly AppDbContext _context;
+        private readonly IMapper _mapper;
+        private readonly ILogger<DriverDashboardService> _logger;
+
+        public DriverDashboardService(AppDbContext context, IAuthenticateService authService, IMapper mapper, ILogger<DriverDashboardService> logger)
+        {
+            _context = context;
+            _authService = authService;
+            _mapper = mapper;
+            _logger = logger;
+        }
+
+        public async Task<List<TaxiBookingViewModel>> GetAllBookingsAsync()
+        {
+
+            var driverId = _authService.GetCurrentUserId();
+
+            if (driverId == null)
+                return new List<TaxiBookingViewModel>();
+
+            var bookings = await _context.TaxiBookings
+            .Where(b => b.Status == ReservationStatus.Pending)
+                .Include(b => b.User)
+                .Include(b => b.Taxi)
+                .ThenInclude(t => t.Driver)
+                .Where(t => t.DriverId == driverId.Value)
+                .ToListAsync();
+
+            return _mapper.Map<List<TaxiBookingViewModel>>(bookings);
+        }
+
+        public async Task<bool> ClaimBookingAsync(int bookingId)
+        {
+            var driverId = _authService.GetCurrentUserId();
+
+            if (driverId == null)
+                return false;
+
+            var booking = await _context.TaxiBookings
+                .Include(b => b.Taxi)
+                .ThenInclude(t => t.Driver)
+                .FirstOrDefaultAsync(b => b.BookingId == bookingId && b.Status == ReservationStatus.Pending);
+
+            if (booking == null || booking.Taxi.DriverId != driverId.Value)
+                return false;
+            booking.Status = ReservationStatus.Confirmed;
+            _context.TaxiBookings.Update(booking);
+            await _context.SaveChangesAsync();
+            return true;
+        }
+
+        public async Task<(bool Success, string Message)> StartTripAsync(int bookingId)
+        {
+            var driverId = _authService.GetCurrentUserId();
+
+            if (driverId == null)
+            {
+                return (false, "Driver is not authenticated.");
+            }
+
+            var booking = await _context.TaxiBookings
+                .Include(b => b.Taxi)
+                .ThenInclude(t => t.Driver)
+                .FirstOrDefaultAsync(b => b.BookingId == bookingId);
+
+            if (booking == null)
+            {
+                return (false, "Booking not found.");
```

Comment:
this and others in this service would be better to return false and add a logger with the message